nautobot/extras/jobs.py

```diff
@@ -104,25 +105,190 @@ class Meta:
         """
 
     def __init__(self):
-        self.logger = logging.getLogger(f"{__name__}.{self.__class__.__name__}")
+        self.logger = get_task_logger(self.__module__)
 
-        self.request = None
         self.active_test = "main"
         self.failed = False
-        self._job_result = None
+        self.job_result = None
 
-        # Compile test methods and initialize results skeleton
-        self.test_methods = []
-
-        for method_name in dir(self):
-            if method_name.startswith("test_") and callable(getattr(self, method_name)):
-                self.test_methods.append(method_name)
+    def __call__(self, *args, **kwargs):
+        # Attempt to resolve serialized data back into original form by creating querysets or model instances
+        # If we fail to find any objects, we consider this a job execution error, and fail.
+        # This might happen when a job sits on the queue for a while (i.e. scheduled) and data has changed
+        # or it might be bad input from an API request, or manual execution.
+        try:
+            deserialized_kwargs = self.deserialize_data(kwargs)
+        except Exception:
+            stacktrace = traceback.format_exc()
+            self.log_failure(f"Error initializing job:\n```\n{stacktrace}\n```")
+        self.active_test = "run"
+        context_class = JobHookChangeContext if isinstance(self, JobHookReceiver) else JobChangeContext
```

Comment:
I'd love for us to rethink this too - can we make JobHookReceiver just "a job whose run() signature matches a certain specific pattern" rather than needing a separate class for this purpose?

Reply:
Agree. I opened #3448 to track that